torch/_inductor/pattern_matcher.py

```diff
@@ -1953,14 +1959,15 @@ def process_arg_fn_impl(
         def run_node(self, n: torch.fx.Node) -> Any:
             rv = super().run_node(n)
             if n.op == "output" and isinstance(rv, tuple):
-                assert len(rv) == len(n.args[0])  # type: ignore[arg-type]
-                for r, arg in zip(rv, n.args[0]):  # type: ignore[arg-type]
+                args = cast(Collection[Any], n.args[0])
```

Comment:
This can just be an isinstance check, Collection is an abc right? Cast should be avoided whenever possible (and isinstance asserts should be instead be used) as it can hide legitimate type errors and help lead to subtle bugs.

Reply:
Also, we make want to make fx.Node.args a TypeVarTuple in a subsequent PR so that it keep tracks of the args typing when possible in the future.

Reply:
@Skylion007 Sure, I'm agreeing with you in general on casts now, changed.

I've found one case so far where I felt they worked better - when we are passing scalar tensors to bits of code that expect plain old Python scalars.